org.eclipse.emfcloud.validation/src/main/java/org/eclipse/emfcloud/validation/ValidationFramework.java

```diff
@@ -0,0 +1,204 @@
+		for (ValidationFilter f : filters) {
+			if (!validationFilterList.contains(f))
+				validationFilterList.add(f);
+		}
+		this.validate();
+	}
+
+	public void removeValidationFilter(List<ValidationFilter> filters)
+			throws IOException, InterruptedException, ExecutionException {
+		for (ValidationFilter filter : filters) {
+			if (validationFilterList.contains(filter))
+				validationFilterList.remove(filter);
+		}
+		this.validate();
+	}
+
+	public void toggleValidationFilter(ValidationFilter filter)
+			throws IOException, InterruptedException, ExecutionException {
+		if (validationFilterList.contains(filter)) {
+			validationFilterList.remove(filter);
+		} else {
+			validationFilterList.add(filter);
+		}
+		this.validate();
+	}
+
+	private void readData(String body) throws IOException {
+		ObjectMapper mapper = new ObjectMapper();
+		mapper.registerModule(new ValidationMapperModule());
+		mapper.registerModule(new ValidationResultModule());
+		JsonNode node = mapper.readTree(body);
+		if (node.get("type").asText().equals("validationResult")) {
+			JsonNode responseData = node.get("data");
+			updateRecentValidationResult(jsonToValidationResultList(mapper, responseData));
+		}
+	}
+
+	private void readConstraintList(String body) throws IOException {
+		ObjectMapper mapper = EMFModule.setupDefaultMapper();
+		mapper.registerModule(new ValidationMapperModule());
+		JsonNode node = mapper.readTree(body);
+		if (node.get("type").asText().equals("success")) {
+			JsonNode listOfElements = node.get("data");
+			// Iterate over all Elements
+			Iterator<String> iterElements = listOfElements.fieldNames();
+			while (iterElements.hasNext()) {
+				String elementKey = iterElements.next();
+				HashMap<String, EMFFacetConstraints> featuresMap = new HashMap<>();
+				JsonNode listOfFeatures = listOfElements.get(elementKey);
+				// Iterate over all Features
+				Iterator<String> iterFeature = listOfFeatures.fieldNames();
+				while (iterFeature.hasNext()) {
+					String featureKey = iterFeature.next();
+					JsonNode facets = listOfFeatures.get(featureKey);
+
+					EMFFacetConstraints emfFacetConstraints = new EMFFacetConstraints(
+							mapper.convertValue(facets, Map.class));
+					featuresMap.put(featureKey, emfFacetConstraints);
+				}
+				this.inputValidationMap.put(elementKey, featuresMap);
+			}
+		}
+	}
+
+	public void updateRecentValidationResult(List<ValidationResult> validationResults) {
+		this.recentValidationResult = validationResults;
+		this.changeListener.changed(validationResults);
+	}
+
+	List<ValidationResult> jsonToValidationResultList(ObjectMapper mapper, JsonNode responseData)
+			throws JsonProcessingException {
+		List<ValidationResult> result = new ArrayList<ValidationResult>();
+		if (!isDiagnosisMessage(responseData.get(ValidationMapperModule.CODE).asInt(),
+				responseData.get(ValidationMapperModule.SOURCE).asText())) {
+			if (responseData.get(ValidationMapperModule.SEVERITY).asInt() != 0) {
```

Comment:
I suggest to extract this also to a helper method

Reply:
@eneufeld Should i just combine this block:
`if (!isDiagnosisMessage(responseData.get(ValidationMapperModule.CODE).asInt(),
				responseData.get(ValidationMapperModule.SOURCE).asText())) {
			if (responseData.get(ValidationMapperModule.SEVERITY).asInt() != 0) {`
			
into something like:
`if(diagnosticIsFiltered(JsonNode responseData){`

Or shall we leave the distinction between the diagnosisMessage and the severity?

Reply:
I'm fine into combining it, if you don't need to filter anything else, 
a better name than `diagnosticIsFiltered` would be great though ;-) . Maybe something like `isErrorDiagnostic` or `isDiagnosticNotOk` ?

Reply:
Ok i updated it, but i also kept `isdiagnosisMessage()` and added `isSeverityOK()` to allow for better readability. Also i readded the needed dependencies in the POM as the maven build was failing for me.